core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -424,20 +425,33 @@ public void onTopologyUpdate(final CacheTopology cacheTopology, final boolean is
          // Remove the transactions whose originators have left the cache.
          // Need to do it now, after we have applied any transactions from other nodes,
          // and after notifyTransactionDataReceived - otherwise the RollbackCommands would block.
-         if (transactionTable != null) {
-            transactionTable.cleanupLeaverTransactions(rpcManager.getTransport().getMembers());
+         try {
+            if (transactionTable != null) {
+               transactionTable.cleanupLeaverTransactions(rpcManager.getTransport().getMembers());
+            }
+         } catch (Exception e) {
+            // Do not fail state transfer when the cleanup fails. See ISPN-7437 for details.
+            log.transactionCleanupError(e);
          }
 
          commandAckCollector.onMembersChange(newWriteCh.getMembers());
 
+         // The rebalance (READ_OLD_WRITE_ALL) is confirmed through notifyEndOfRebalanceIfNeeded
+         // and STABLE does not have to be confirmed at all
```

Comment:
I was considering using a `invokeRemotelyAsync`, but as the topology confirmation is sent from different parts, that would mean creating an uncompleted CF, registering it for given topology somewhere (`LocalCacheStatus`?) and something similar on the coordinator side. I've left that path after a while of refactoring.